src/ripple/protocol/impl/Keylet.cpp

```diff
@@ -23,17 +23,16 @@
 namespace ripple {
 
 bool
-Keylet::check(SLE const& sle) const
+Keylet::check(STLedgerEntry const& sle) const
 {
+    assert(sle.getType() != ltANY || sle.getType() != ltCHILD);
+
     if (type == ltANY)
         return true;
-    if (type == ltINVALID)
-        return false;
+
     if (type == ltCHILD)
-    {
-        assert(sle.getType() != ltDIR_NODE);
```

Comment:
What are the cases where we expect `check` to fail? I think they are: 1) logic errors, 2) corrupt data, 3) hash collisions. If that's true, maybe we should assert of any `check` failure? At any rate, I'm OK removing the assert, it does seem like it should either always assert on any `check` failure or never assert.